kolibri/plugins/facility/assets/src/views/FacilityConfigPage/EditFacilityNameModal.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+
+  <KModal
+    :title="$tr('title')"
+    :submitText="coreString('saveAction')"
+    :cancelText="coreString('cancelAction')"
+    @submit="handleSubmit"
+    @cancel="$emit('cancel')"
+  >
+    <div>
+      <p>{{ $tr('renameFacilityExplanation') }}</p>
+      <KTextbox
+        ref="name"
+        v-model.trim="name"
+        type="text"
+        :label="coreString('facilityName')"
+        :autofocus="true"
+        :invalid="nameIsInvalid"
+        :invalidText="nameIsInvalidText"
+        :maxlength="50"
+        @blur="nameBlurred = true"
+        @input="handleInput"
+      />
+    </div>
+  </KModal>
+
+</template>
+
+
+<script>
+
+  import commonCoreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
+  import { mapState } from 'vuex';
+
+  export default {
+    name: 'EditFacilityNameModal',
+    mixins: [commonCoreStrings],
+    props: {
+      facilityName: {
+        type: String,
+        required: false,
+      },
+    },
+    data() {
+      return {
+        name: this.facilityName,
+        nameBlurred: false,
+        formSubmitted: false,
+        isDuplicated: false,
+      };
+    },
+    computed: {
+      ...mapState('facilityConfig', ['facilities']),
+      nameIsInvalidText() {
+        if (this.name === '') {
+          return this.coreString('requiredFieldError');
+        }
+        if (this.isDuplicated) return this.coreString('facilityDuplicated');
+        return '';
+      },
+      nameIsInvalid() {
+        return Boolean(this.nameIsInvalidText || this.isDuplicated);
```

Comment:
Can remove `this.isDuplicated` since `this.nameIsInvalidText` should be truthy iff `this.isDuplicated` is true.

```suggestion
        return Boolean(this.nameIsInvalidText);
```

Reply:
Done